### PR TITLE
feat: add dragStart and dragEnd events (close #222)

### LINKED ISSUE #222: Drag start and drag end output event
* **I'm submitting a ...**
  - [ ] bug report
  - [x] feature request
  - [ ] support request => Please do not submit support request here, see note at the top of this template.

* **Do you want to request a *feature* or report a *bug*?**
Feature

### PR DESCRIPTION
**Pull request by dintecom**
- [x] The commit message follows our guidelines
- [ ] Tests for the changes have been added (for bug fixes / features)
- [x] Docs have been added / updated (for bug fixes / features)
- [ ] Unit tests are passing `ng test`
- [x] Lint tests are passing `ng lint`

* **What kind of change does this PR introduce?** (Bug fix, feature, docs update, ...)
Add dragStart and dragEnd events (close #222)

* **What is the current behavior?** (You can also link to an open issue here)

* **What is the new behavior (if this is a feature change)?**

* **Does this PR introduce a breaking change?** (What changes might users need to make in their application due to this PR?)

* **Other information**:

### CODE REVIEW

File: projects/ngx-drag-scroll/src/lib/ngx-drag-scroll.component.ts [hunk 1]
```diff
@@ -421,6 +436,15 @@ export class DragScrollComponent implements OnDestroy, AfterViewInit, OnChanges,
...
 
+  private _setIsDragging(value: boolean) {
+    if (this._isDragging === value) {
+      return;
+    }
+
+    this._isDragging = value;
+    value ? this.dragStart.emit() : this.dragEnd.emit();
+  }
+
```
Comment (bfwg) [inline projects/ngx-drag-scroll/src/lib/ngx-drag-scroll.component.ts]: How do you feel about using a private setter for `isDragging`. Is matches the getter/setter convension.
```suggestion
  private set isDragging(value: boolean) {
    if (this._isDragging === value) {
      return;
    }

     this._isDragging = value;
    value ? this.dragStart.emit() : this.dragEnd.emit();
  }
```

Comment (dintecom) [reply projects/ngx-drag-scroll/src/lib/ngx-drag-scroll.component.ts]: Thanks for the tip. In older versions of typescript this did not work and I made a mistake out of habit.

Comment (bfwg) [reply projects/ngx-drag-scroll/src/lib/ngx-drag-scroll.component.ts]: All good. Should the 'I' in `set IsDragging` be lowercase? Everything else looks good!

Comment (bfwg) [reply projects/ngx-drag-scroll/src/lib/ngx-drag-scroll.component.ts]: Oh, can you also merge your PR with develop and resolve the conflicts please?

Comment (dintecom) [reply projects/ngx-drag-scroll/src/lib/ngx-drag-scroll.component.ts]: Unfortunately, this does not work in this version of typescript (error TS2379: Getter and setter accessors do not agree in visibility).

Comment (dintecom) [reply projects/ngx-drag-scroll/src/lib/ngx-drag-scroll.component.ts]: This feature has long been asked from Microsoft https://github.com/Microsoft/TypeScript/issues/2845 , but it never did.

**Pull request merged**